torch/distributions/transformed_distribution.py

```diff
@@ -87,21 +88,35 @@ def log_prob(self, value):
                                    event_dim - len(self.base_dist.event_shape))
         return log_prob
 
+    def _monotonize_cdf(self, value):
+        """
+        This conditionally flips ``value -> 1-value`` to ensure :meth:`cdf` is
+        monotone increasing.
+        """
+        sign = 1
+        for transform in self.transforms:
+            sign = sign * transform.sign
+        if sign is 1:
```

Comment:
why do we need this?

Reply:
it's just an easy optimization

Reply:
But isn't performing twice as many loop iterations?

Reply:
This is a tiny loop, typically of size 1. We could fuse the loops by inlining this function into `.cdf()` and `.icdf()` but it seems cleaner to me to abstract it out into a helper so that `.icdf()` and `.cdf()` are easier to read.